provider/providertest/integration.go

```diff
@@ -0,0 +1,422 @@
+			}
+		}
+	}
+	return nil
+}
+
+// compareDataWithExpected - runs through expected values and checks if they are satisfied with received data
+func compareDataWithExpected(expected []ExpectedValue, received []map[string]interface{}) error {
+	var errors []error
+	// clone []map that will be compared
+	toCompare := make([]map[string]interface{}, len(received))
+	for i, row := range received {
+		toCompare[i] = make(map[string]interface{})
+		for key, value := range row {
+			toCompare[i][key] = value
+		}
+	}
+
+	for _, verification := range expected {
+		found := 0
+		for i := 0; i < len(toCompare); i++ {
+			if toCompare[i] == nil {
+				continue // this row is already verified - skip
+			}
+			err := compareData(verification.Data, toCompare[i])
+			if err == nil {
+				toCompare[i] = nil // row passed verification - it won't be used
+				found++
+			} else {
+				errors = append(errors, err)
+			}
+
+		}
+		if verification.Count != found {
+			return fmt.Errorf("expected to have %d but got %d entries with one of the %v\nerrors: %v", verification.Count, found, verification.Data, errors)
+		}
+	}
+	return nil
+}
+
+// compareData - checks if the second argument has all the entries of the first argument. arguments are jsons - map[string]interface{}
+func compareData(verification, row map[string]interface{}) error {
+	for k, v := range verification {
+		diff := deep.Equal(row[k], v)
+		if diff != nil {
+			return fmt.Errorf("%+v", diff)
+		}
+	}
+	return nil
+}
+
+// simplifyString - prepares the string to be used in resources names
+func simplifyString(in string) string {
+	return strings.ToLower(tfVarRegex.ReplaceAllString(in, ""))
+}
+
+// getDataFromRow - reads the row from db into an array jsons: []map[string]interface{}
+func getDataFromRow(row pgx.Row) ([]map[string]interface{}, error) {
+	var resp []map[string]interface{}
+	var data string
+
+	if err := row.Scan(&data); err != nil {
+		return nil, err
+	}
+	if err := json.Unmarshal([]byte(data), &resp); err != nil {
+		return nil, err
+	}
+	return resp, nil
+}
+
+// copyTfFiles - copies tf files that are related to current test
+func copyTfFiles(name string) (string, error) {
+	workdir := filepath.Join(tfDir, name)
+	if _, err := os.Stat(workdir); os.IsNotExist(err) {
+		if err := os.MkdirAll(workdir, os.ModePerm); err != nil {
+			return workdir, err
+		}
+	} else {
+		return workdir, err
+	}
+
+	err := cp(filepath.Join(tfOrigin, name+".tf"), filepath.Join(workdir, name+".tf"))
+	if err != nil {
+		return workdir, err
+	}
+
+	err = cp(filepath.Join(tfOrigin, "variables.tf"), filepath.Join(workdir, "variables.tf"))
+	if err != nil {
+		return workdir, err
+	}
+
+	err = cp(filepath.Join(tfOrigin, "provider.tf"), filepath.Join(workdir, "provider.tf"))
+	if err != nil {
+		return workdir, err
+	}
+
+	err = cp(filepath.Join(tfOrigin, "versions.tf"), filepath.Join(workdir, "versions.tf"))
+	if err != nil {
+		return workdir, err
+	}
```

Comment:
create a for loop for this, with []string array of files, to reduce code duplication